lib/packExternalModules.js

```diff
@@ -193,22 +192,49 @@ module.exports = {
       // (1.a) Install all needed modules
       const compositeModulePath = path.join(this.webpackOutputPath, 'dependencies');
       const compositePackageJson = path.join(compositeModulePath, 'package.json');
-      this.serverless.utils.writeFileSync(compositePackageJson, '{}');
 
-      this.serverless.cli.log('Packing external modules: ' + compositeModules.join(', '));
+      // (1.a.1) Create a package.json
+      const compositePackage = {
+        name: this.serverless.service.service,
+        version: '1.0.0',
+        description: `Packaged externals for ${this.serverless.service.service}`,
+        private: true
+      };
+      _.forEach(compositeModules, compositeModule => {
+        const splitModule = _.split(compositeModule, '@');
+        // If we have a scoped module we have to re-add the @
+        if (_.startsWith(compositeModule, '@')) {
+          splitModule.splice(0, 1);
+          splitModule[0] = '@' + splitModule[0];
+        }
+        const moduleVersion = _.join(_.tail(splitModule), '@');
+        _.set(compositePackage, `dependencies.${_.first(splitModule)}`, moduleVersion);
```

Comment:
@silver2k This is the issue. The assembly of the dependency name breaks here if the dependency includes a `.`.

It is handled as if it was a nested property. I will provide a fix asap and release 3.1.2.